core/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -41,13 +44,7 @@ private[spark] trait Logging {
   }
 
   // Method to get or create the logger for this object
-  protected def log: Logger = {
-    if (log_ == null) {
-      initializeLogIfNecessary(false)
-      log_ = LoggerFactory.getLogger(logName)
-    }
-    log_
-  }
+  protected def log: Logger = log_
```

Comment:
Good point. I think that can be removed. I don't see the point of it, as neither it nor its subclass even logs anything directly?